File: 11. Container With Most Water/Solution.java
class Solution {
    public int maxArea(int[] height) {
        int len = height.length;
        int res = 0;
        int right = len-1;
        int left = 0;
        while(left != right){            
            // System.out.println(left+" - "+right);
            int temp = Math.min(height[left], height[right]) * (right-left);
            // System.out.println(temp);
            if(height[left] < height[right]) {
                left++;
            }else {
                right--;
            }
            res = Math.max(temp, res);
        }
    
        return res;
    }
}

File: 11. Container With Most Water/Solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        area = 0
        while left != right:
            area = max(area, min(height[left], height[right]) * (right-left))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return area
